src/hotspot/share/cds/archiveHeapWriter.cpp

```diff
@@ -608,50 +504,47 @@ void ArchiveHeapWriter::mark_native_pointer(oop src_obj, int field_offset) {
   }
 }
 
-ArchiveHeapBitmapInfo ArchiveHeapWriter::compute_ptrmap(bool is_open) {
+void ArchiveHeapWriter::compute_ptrmap(ArchiveHeapInfo* heap_info) {
   int num_non_null_ptrs = 0;
-  Metadata** bottom = (Metadata**) (is_open ? _requested_open_region_bottom: _requested_closed_region_bottom);
-  Metadata** top = (Metadata**) (is_open ? _requested_open_region_top: _requested_closed_region_top); // exclusive
-  ResourceBitMap ptrmap(top - bottom);
+  Metadata** bottom = (Metadata**) _requested_bottom;
+  Metadata** top = (Metadata**) _requested_top; // exclusive
+  heap_info->ptrmap()->resize(top - bottom);
 
+  BitMap::idx_t max_idx = 32; // paranoid - don't make it too small
   for (int i = 0; i < _native_pointers->length(); i++) {
     NativePointerInfo info = _native_pointers->at(i);
     oop src_obj = info._src_obj;
     int field_offset = info._field_offset;
     HeapShared::CachedOopInfo* p = HeapShared::archived_object_cache()->get(src_obj);
-    if (p->in_open_region() == is_open) {
-      // requested_field_addr = the address of this field in the requested space
-      oop requested_obj = requested_obj_from_buffer_offset(p->buffer_offset());
-      Metadata** requested_field_addr = (Metadata**)(cast_from_oop<address>(requested_obj) + field_offset);
-      assert(bottom <= requested_field_addr && requested_field_addr < top, "range check");
-
-      // Mark this field in the bitmap
-      BitMap::idx_t idx = requested_field_addr - bottom;
-      ptrmap.set_bit(idx);
-      num_non_null_ptrs ++;
-
-      // Set the native pointer to the requested address of the metadata (at runtime, the metadata will have
-      // this address if the RO/RW regions are mapped at the default location).
-
-      Metadata** buffered_field_addr = requested_addr_to_buffered_addr(requested_field_addr);
-      Metadata* native_ptr = *buffered_field_addr;
-      assert(native_ptr != nullptr, "sanity");
-
-      address buffered_native_ptr = ArchiveBuilder::current()->get_buffered_addr((address)native_ptr);
-      address requested_native_ptr = ArchiveBuilder::current()->to_requested(buffered_native_ptr);
-      *buffered_field_addr = (Metadata*)requested_native_ptr;
+    // requested_field_addr = the address of this field in the requested space
+    oop requested_obj = requested_obj_from_buffer_offset(p->buffer_offset());
+    Metadata** requested_field_addr = (Metadata**)(cast_from_oop<address>(requested_obj) + field_offset);
+    assert(bottom <= requested_field_addr && requested_field_addr < top, "range check");
+
+    // Mark this field in the bitmap
+    BitMap::idx_t idx = requested_field_addr - bottom;
+    heap_info->ptrmap()->set_bit(idx);
+    num_non_null_ptrs ++;
+
+    if (max_idx < idx) {
```

Comment:
Is there a built in min() function we can use here? Maybe std::min()?

Reply:
Updated with the `MAX2()` macro.